tuning/autotune.py

```diff
@@ -25,10 +29,26 @@
 
 # Default values for max number of workers
 DEFAULT_MAX_CPU_WORKERS = (
-    multiprocessing.cpu_count()//2 
+    multiprocessing.cpu_count() // 2
 )  # the actual amount of worker that will be generated = max(min(max_cpu_workers//2, len(task_list)), 1)
 """note: Do not use all CPU cores"""
 
+@dataclass
+class TuningCandidate:
+    candidate_id: int
+    mlir_path: str = None
+    mlir_config_path: str = None
+    configuration: tune.Configuration = None
+    compilation_successful: bool = None
+    compiled_vmfb_path: str = None
+    first_benchmark_time: float = None
+    unet_candidate_path: str = None
+    unet_vmfb_hash: str = None
+    unet_benchmark_time: float = None
+
+
+Candidate_Tracker: list[TuningCandidate] = []
```

Comment:
Can we avoid using global variables? These make it difficult to understand where the program state changes and are hard to test.